File: FoodForecastingAPI/FoodForecastingAPI/Models/SurplusFood.cs
namespace FoodForecastingAPI.Models
{
    public class SurplusFood
    {
        public int Id { get; set; }

        public DateTime SurplusDate { get; set; }

        public string MealType { get; set; } = "";

        public string MealName { get; set; } = "";

        public int ProducedQuantity { get; set; }

        public int ConsumedQuantity { get; set; }

        public int SurplusQuantity { get; set; }

        public string Status { get; set; } = "Available";
    }
}

File: food-forecasting-ui/src/app/services/surplus-food.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface SurplusFood {
  id?: number;
  surplusDate: string;
  mealType: string;
  mealName: string;
  producedQuantity: number;
  consumedQuantity: number;
  surplusQuantity: number;
  status: string;
}

@Injectable({
  providedIn: 'root'
})
export class SurplusFoodService {

  private apiUrl = 'http://localhost:5003/api/SurplusFoods';

  constructor(private http: HttpClient) {}

  getSurplusFoods(): Observable<SurplusFood[]> {
    return this.http.get<SurplusFood[]>(this.apiUrl);
  }

  addSurplusFood(surplusFood: SurplusFood): Observable<SurplusFood> {
    return this.http.post<SurplusFood>(
      this.apiUrl,
      surplusFood
    );
  }

  deleteSurplusFood(id: number): Observable<void> {
    return this.http.delete<void>(
      `${this.apiUrl}/${id}`
    );
  }
}